daemon/update_linux.go

```diff
@@ -15,7 +15,9 @@ func toContainerdResources(resources container.Resources) libcontainerd.Resource
 	r.CpuQuota = uint64(resources.CPUQuota)
 	r.CpusetCpus = resources.CpusetCpus
 	r.CpusetMems = resources.CpusetMems
-	r.MemoryLimit = uint64(resources.Memory)
+	if resources.Memory != 0 {
+		r.MemoryLimit = uint64(resources.Memory)
```

Comment:
I removed `if` statement as it's not needed, I think that was left over from master rebase.
The type change requires change to `containerd' types here: https://github.com/docker/containerd/blob/master/api/grpc/types/api.pb.go#L469

I can update that too, it makes sense to use `int64` instead. Please give a thumbs up if my proposed change is correct.